File: powerfulsharedpreferences/src/main/java/com/stefanosiano/powerfulsharedpreferences/PrefContainer.java
package com.stefanosiano.powerfulsharedpreferences;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.SparseArray;

/**
 * Created by stefano on 11/23/17.
 */

class PrefContainer {
    SharedPreferences sharedPreferences;
    boolean useCrypter;
    String name;
    int mode;

    PrefContainer(boolean useCrypter, String name, int mode) {
        this.useCrypter = useCrypter;
        this.name = name;
        this.mode = mode;
    }
    void build(Context context){
        sharedPreferences = context.getApplicationContext().getSharedPreferences(name, mode);
    }
}

File: powerfulsharedpreferences/src/main/java/com/stefanosiano/powerfullibraries/sharedpreferences/PrefContainer.kt
package com.stefanosiano.powerfullibraries.sharedpreferences

import android.content.Context
import android.content.SharedPreferences


internal class PrefContainer(var useCrypter: Boolean, var name: String, var mode: Int) {
    var sharedPreferences: SharedPreferences? = null
    fun build(context: Context) { sharedPreferences = context.applicationContext.getSharedPreferences(name, mode) }
}